packages/transaction-controller/src/gas-flows/LineaGasFeeFlow.ts

```diff
@@ -0,0 +1,209 @@
+import {
+  ChainId,
+  gweiDecToWEIBN,
+  hexToBN,
+  query,
+  toHex,
+} from '@metamask/controller-utils';
+import type EthQuery from '@metamask/eth-query';
+import type { GasFeeEstimates } from '@metamask/gas-fee-controller';
+import { GAS_ESTIMATE_TYPES } from '@metamask/gas-fee-controller';
+import { createModuleLogger, type Hex } from '@metamask/utils';
+import type { BN } from 'ethereumjs-util';
+
+import { projectLogger } from '../logger';
+import type {
+  GasFeeFlow,
+  GasFeeFlowRequest,
+  GasFeeFlowResponse,
+  TransactionMeta,
+} from '../types';
+import { DefaultGasFeeFlow } from './DefaultGasFeeFlow';
+
+type LineaEstimateGasResponse = {
+  baseFeePerGas: Hex;
+  priorityFeePerGas: Hex;
+};
+
+type FeesByLevel = {
```

Comment:
It seems like we may be mixing up "Level" and "Priority" in this PR. Could you check and make sure we're consistent with these terms?

Reply:
The term we want to use is `Level` as I think priority is ambiguous given `priority` in `maxPriorityFeePerGas`.

The new type is called `GasEstimateLevel`.